src/mds/MDSContext.h

```diff
@@ -187,6 +188,15 @@ class MDSGather : public C_GatherBase<MDSInternalContextBase, MDSInternalContext
   MDSRank *get_mds() override {return NULL;}
 };
 
+class MDSContextSleeper : public MDSInternalContext, public C_SaferCond {
+public:
+  MDSContextSleeper() = delete;
+  MDSContextSleeper(const std::string &name, MDSRank *mds) :
+    MDSInternalContext(mds), C_SaferCond(name) {}
+  void finish(int r) {
+    return;
```

Comment:
Should I implement this? Do we have an example?

By the way, just in case if this comes in handy, ACKs are received by the time control reaches this line; observed during a gdb session.